lib/ObjWriter/objwriter.cpp

```diff
@@ -631,6 +730,37 @@ extern "C" void EmitDebugFunctionInfo(ObjectWriter *OW,
   OST.EmitCVLinetableDirective(OW->FuncId++, Fn, FnEnd);
 }
 
+extern "C" void EmitDebugFunctionInfo(ObjectWriter *OW,
+                                      const char *FunctionName,
+                                      int FunctionSize) {
+  assert(OW && "ObjWriter is null");
+  auto *AsmPrinter = &OW->getAsmPrinter();
+  auto &OST = static_cast<MCObjectStreamer &>(*AsmPrinter->OutStreamer);
+  MCContext &OutContext = OST.getContext();
+  const MCObjectFileInfo *MOFI = OutContext.getObjectFileInfo();
+
+  if (MOFI->getObjectFileType() == MOFI->IsCOFF) {
+    EmitCVDebugFunctionInfo(OW, FunctionName, FunctionSize);
+  } else {
+    // TODO: Should convert this for non-Windows.
+  }
+}
+
+extern "C" void EmitDebugVar(ObjectWriter *OW, char *Name, int TypeIndex,
+                             bool IsParm, int RangeCount,
+                             ICorDebugInfo::NativeVarInfo *Ranges) {
+  assert(OW && "ObjWriter is null");
+  assert(RangeCount != 0);
+  DebugVarInfo NewVar(Name, TypeIndex, IsParm);
+  OW->DebugVarInfos.push_back(NewVar);
+  DebugVarInfo &Var = OW->DebugVarInfos.back();
```

Comment:
That may result in copying a large array from NewVar to &Var, due to how verctor.copy is implemented.